src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -480,7 +480,10 @@ def range_statement(start: int, end: int, step: int, column_name: str) -> str:
 
 
 def schema_query_for_values_statement(output: List[Attribute]) -> str:
-    cells = [schema_expression(attr.datatype, attr.nullable) for attr in output]
+    cells = [
+        schema_expression(cast(DataType, attr.datatype), attr.nullable)
```

Comment:
`Attribute.datatype` has type `Optional[DataType]`, which isn't accepted here. Usually in this PR I've dealt with this by writing something like `assert attr.datatype is not None` which satisfies the type checker. I use `cast` here so that the list comprehension can stay.

However, I've taken a closer look, and the only reason for the type being optional is so that it can default to None, which is only useful in one place where the data type is not provided: https://github.com/snowflakedb/snowpark-python/blob/2243bb617b72982ded33e6326cc9bae76ec47205/src/snowflake/snowpark/_internal/analyzer/select_statement.py#L1096-L1099

So maybe a better solution is to make that argument required and provide a dummy datatype in that place.